op-batcher/batcher/espresso.go

```diff
@@ -21,43 +22,529 @@ import (
+// for receipt verifications.
+type espressoVerifyReceiptJobAttempt struct {
+	job  espressoVerifyReceiptJob
+	resp chan espressoVerifyReceiptJobResponse
+}
+
+// espressoTransactionSubmitter is a struct that holds the state that governs
+// the worker queue processing details for submitting transactions to Espresso
+// without spawning arbitrarily many goroutines.
+type espressoTransactionSubmitter struct {
+	ctx                      context.Context
+	wg                       *sync.WaitGroup
+	submitJobQueue           chan espressoSubmitTransactionJob
+	submitRespQueue          chan espressoSubmitTransactionJobResponse
+	submitWorkerQueue        chan chan espressoTransactionJobAttempt
+	verifyReceiptJobQueue    chan espressoVerifyReceiptJob
+	verifyReceiptRespQueue   chan espressoVerifyReceiptJobResponse
+	verifyReceiptWorkerQueue chan chan espressoVerifyReceiptJobAttempt
+	espresso                 espressoClient.EspressoClient
+}
+
+// EspressoTransactionSubmitterConfig is a configuration struct for the
+// EspressoTransactionSubmitter. It contains the configurable details for
+// creating the EspressoTransactionSubmitter.
+type EspressoTransactionSubmitterConfig struct {
+	Ctx                                context.Context
+	EspressoClient                     espressoClient.EspressoClient
+	Wg                                 *sync.WaitGroup
+	SubmitJobQueueCapacity             int
+	SubmitResponseQueueCapacity        int
+	VerifyReceiptJobQueueCapacity      int
+	VerifyReceiptResponseQueueCapacity int
+}
+
+// EspressoTransactionSubmitterOption is a function that can be used to
+// configure the EspressoTransactionSubmitterConfig.
+type EspressoTransactionSubmitterOption func(*EspressoTransactionSubmitterConfig)
+
+// WithEspressoClient is an option that can be used to set the Espresso client
+// for the EspressoTransactionSubmitterConfig.
+func WithContext(ctx context.Context) EspressoTransactionSubmitterOption {
+	return func(config *EspressoTransactionSubmitterConfig) {
+		config.Ctx = ctx
 	}
+}
 
-	timer := time.NewTimer(transactionFetchTimeout)
-	defer timer.Stop()
+// WithEspressoClient is an option that can be used to set the Espresso client
+// for the EspressoTransactionSubmitterConfig.
+func WithEspressoClient(client espressoClient.EspressoClient) EspressoTransactionSubmitterOption {
+	return func(config *EspressoTransactionSubmitterConfig) {
+		config.EspressoClient = client
+	}
+}
 
-	ticker := time.NewTicker(transactionFetchInterval)
-	defer ticker.Stop()
+// WithWaitGroup is an option that can be used to set the wait group
+// for the EspressoTransactionSubmitterConfig.
+func WithWaitGroup(wg *sync.WaitGroup) EspressoTransactionSubmitterOption {
+	return func(config *EspressoTransactionSubmitterConfig) {
+		config.Wg = wg
+	}
+}
+
+// NewEspressoTransactionSubmitter creates a new EspressoTransactionSubmitter
+// with the given context and espresso client.  It will create a new
```

Comment:
> It will create a new

Looks like the sentence is not finished.

Reply:
Addressed in db23892adc18ad7fe5ff7c51cfffb1203b8cb3ac.